### PR TITLE
Skript Enhancements

### PR DESCRIPTION
**Pull request by Olyno**
### Description

All referenced issues in #2506 have been fixed in this pull request.

---
**Target Minecraft Versions:**     none
**Requirements:**     none
**Related Issues:**     #2506

### CODE REVIEW

File: src/main/resources/scripts/vanilla gui.sk [hunk 1]
```diff
@@ -0,0 +1,15 @@
+#
+# A vanilla example of a gui creation using Skript only
+# Requiert Skript 2.3+
```
Comment (Matocolotoe) [inline src/main/resources/scripts/vanilla gui.sk]:
```suggestion
# Requires Skript 2.3+
```
Le français n'est jamais très loin xd

File: src/main/resources/scripts/teleport with compass.sk [hunk 2]
```diff
@@ -3,14 +3,8 @@
 # with a compass like in WorldEdit.
 #
 
-aliases:
-	# blocks without collision as of Minecraft 1.5 (excluding fire, nether portal & end portal)
-	nonsolid = 0, 6, 27, 28, 30, 31, 32, 37, 38, 39, 40, 50, 55, 59, 63, 64, 65, 66, 68, 69, 70, 71, 72, 75, 76, 77, 78:0, 83, 85:4-7, 104, 105, 106, 111, 115, 131, 132, 141, 142, 143, 147, 148, 157
-
 on rightclick with compass:
 	player has permission "skript.teleport"
 	loop blocks above targeted block:
```
Comment (Matocolotoe) [inline src/main/resources/scripts/teleport with compass.sk]:
```suggestion
	loop blocks above targeted block [where input isn't solid]:
```

File: src/main/resources/scripts/vanilla gui.sk [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+#
+
+command /gui:
+	trigger:
+		open chest inventory with 6 row with name "Gui Test" to player
```
Comment (Matocolotoe) [inline src/main/resources/scripts/vanilla gui.sk]:
```suggestion
		open chest inventory with 6 rows with name "Gui Test" to player
```

File: src/main/resources/scripts/homes.sk [hunk 4]
```diff
@@ -1,22 +1,30 @@
...
+		message "Set your home <green>%arg-1%<reset> to <grey>%location of player%<reset>"
 
-command /home:
+command /delhome <string>:
+	description: Delete your home
+	permission: skript.home
+	executable by: players
+	trigger:
+		clear {homes::%uuid of player%::%arg-1%}
+		message "Delete your home <green>%arg-1%<reset>"
```
Comment (Matocolotoe) [inline src/main/resources/scripts/homes.sk]:
```suggestion
		message "Deleted your home <green>%arg-1%<reset>"
```

File: src/main/resources/scripts/teleport with compass.sk (hunk 2 above)
Comment (FranKusmiruk) [reply src/main/resources/scripts/teleport with compass.sk]: I'd rather use a simple if check rather than the where filter, it'd confuse new people.

Comment (Matocolotoe) [reply src/main/resources/scripts/teleport with compass.sk]: true

**Comment by Wealthyturtle** [comment]
Should probably configure your git to not treat every file as being completely different so we can actually tell what was modified :P

**Comment by Matocolotoe** [comment]
The files changed section is displayed by GitHub itself

**Comment by Wealthyturtle** [comment]
According to https://git-scm.com/docs/git-diff#Documentation/git-diff.txt---diff-algorithmpatienceminimalhistogrammyers you can?

**Comment by Matocolotoe** [comment]
Didn't know that was possible but it still seems weird 🤔

File: src/main/resources/scripts/teleport with compass.sk (hunk 2 above)
Comment (Olyno) [reply src/main/resources/scripts/teleport with compass.sk]: But how ``nonsolid`` blocks are usefull here? I mean, I did tests without this condition and it works really well.

Comment (Matocolotoe) [reply src/main/resources/scripts/teleport with compass.sk]: Does it not prevent the player from being teleported inside a block ?
Also I think you should format the condition with `if` like you did in other examples

Comment (Olyno) [reply src/main/resources/scripts/teleport with compass.sk]: When you use ``if`` word, you are supposed to follow with the ``else`` part, or you just use it as "code block" so here you are not supposed to use the ``if`` word because it's an "affirmative" condition (so if this condition is not respected, it will not continue the code)

File: docs/README.md [hunk 5]
```diff
@@ -1,28 +1,51 @@
...
+## Export Documentation
 
-Patterns have syntax of ${pattern_here}. For example, ${skript.version} is replaced with
-current Skript version. Please see below for more...
+To export the documentation, run these commands in your terminal:
 
-You can also include other files by using ${include <filename>}. Just please make
-sure that those included files don't have tags which are not allowed in position
-where include is called.
+**Using NPM:**
```
Comment (bensku) [inline docs/README.md]: People exporting documentation should not be required to install anything in addition to Minecraft server + Java.

File: docs/README.md [hunk 6]
```diff
@@ -1,28 +1,51 @@
...
-1. HTML, CSS and (possible) Javascript code to create website out of these
-2. Clear tutorials, not just "you can check the syntax pattern"
-3. Examples explained, if needed
+What the website uses:
 
-When generating final result, each HTML file is surrounded by template.html,
-which provides head element, navigation bar and so on.
+ * Bulma
+ * FontAwesome
+ * Svelte
```
Comment (bensku) [inline docs/README.md]: Documentation website must work without Javascript, or at very least there must be alternative that works.

File: docs/static/bulma.css [hunk 7]
```diff
@@ -0,0 +1 @@
...636 30%,#dbdbdb 30%)}.progress.is-primary::-webkit-progress-value{background-color:#00d1b2}.progress.is-primary::-moz-progress-bar{background-color:#00d1b2}.progress.is-primary::-ms-fill{background-color:#00d1b2}.progress.is-primary:indeterminate{background-image:linear-gradient(to right,#00d1b2 30%,#dbdbdb 30%)}.progress.is-link::-webkit-progress-value{background-color:#3273dc}.progress.is-link::-moz-progress-bar{background-color:#3273dc}.progress.is-link::-ms-fill{background-color:#3273dc}.progress.is-link:indeterminate{background-image:linear-gradient(to right,#3273dc 30%,#dbdbdb 30%)}.progress.is-info::-webkit-progress-value{background-color:#209cee}.progress.is-info::-moz-progress-bar{background-color:#209cee}.progress.is-info::-ms-fill{background-color:#209cee}.progress.is-info:indeterminate{background-image:linear-gradient(to right,#209cee 30%,#dbdbdb 30%)}.progress.is-success::-webkit-progress-value{background-color:#23d160}.progress.is-success::-moz-progress-bar{background-color:#23d160}.progress.is-success::-ms-fill{background-color:#23d160}.progress.is-success:indeterminate{background-image:linear-gradient(to right,#23d160 30%,#dbdbdb 30%)}.progress.is-warning::-webkit-p
```
Comment (bensku) [inline docs/static/bulma.css]: If you're using NPM, why is this included in repo?

Comment (Olyno) [reply docs/static/bulma.css]: For the style of the website?

File: docs/README.md (hunk 6 above)
Comment (Olyno) [reply docs/README.md]: Modern websites use javascript. What's the problem with javascript?

Comment (bensku) [reply docs/README.md]: It is not necessary for static pages. For dynamic features such as search, there is no alternative.

File: docs/static/bulma.css (hunk 7 above)
Comment (bensku) [reply docs/static/bulma.css]: I meant, could you not fetch Bulma using NPM?

File: docs/README.md (hunk 6 above)
Comment (Olyno) [reply docs/README.md]: Agree with you. But modern technologies (as react, and here svelte) use javascript to create modules. These modules are render on the website directly. So without javascript, Svelte, react and more component technologies will not work.

File: docs/static/bulma.css (hunk 7 above)
Comment (Olyno) [reply docs/static/bulma.css]: Sure we can. I just found better to use a static file. Using Bulma's css as static file (and not npm) we allow anyone to change the style of the website without using npm commands.

Comment (Olyno) [reply docs/static/bulma.css]: Sure we can. I just found better to use a static file. Using Bulma's css as static file (and not npm) we allow anyone to change the style of the website without using npm commands.

File: src/main/resources/scripts/teleport with compass.sk (hunk 2 above)
Comment (Matocolotoe) [reply src/main/resources/scripts/teleport with compass.sk]: Yeah that's what I do but I would write it there as a convention, in Java or Python you cannot omit it

File: src/main/resources/scripts/simple join and leave message.sk [hunk 8]
```diff
@@ -0,0 +1,27 @@
+#
+# A join and leave message.
+#
+
+command /set <string> message <text>:
```
Comment (Matocolotoe) [inline src/main/resources/scripts/simple join and leave message.sk]:
```suggestion
command /set <text> message <text>:
```

File: src/main/resources/scripts/simple join and leave message.sk [hunk 9]
```diff
@@ -0,0 +1,27 @@
...
+	permission: skript.set.join message
+	description: Sets message when player joins
+	trigger:
+		if argument-1 is "join" or "leave":
+			set {custom messages::%argument-1%} to argument-2
+			message "Set the %argument-1% message to '%argument-2%<reset>'"
+		else:
+			message "Only 'join' and 'leave' messages are available here."
+
+command /show <string> message:
```
Comment (Matocolotoe) [inline src/main/resources/scripts/simple join and leave message.sk]:
```suggestion
command /show <text> message:
```

File: src/main/resources/scripts/simple join and leave message.sk (hunk 8 above)
Comment (Olyno) [reply src/main/resources/scripts/simple join and leave message.sk]: Should be better to use string instead of text type right?

File: docs/src/utils/developers.json [hunk 10]
```diff
@@ -0,0 +1,34 @@
...
+        "avatar": "https://avatars2.githubusercontent.com/u/12614053?s=400&v=4",
+        "roles": [
+            "Skript Developer"
+        ]
+    },
+    "TheBentoBox": {
+        "avatar": "https://avatars3.githubusercontent.com/u/6902229?s=400&v=4",
+        "roles": [
+            "Issue tracker manager",
+            "Aliases developer"
```
Comment (ShaneBeee) [inline docs/src/utils/developers.json]: Consistency seems a little off in terms of casing. 
You have `Skript Developer` and then `Skript maintainer`
I recommend sticking with upper casing for each word, but thats just my opinion. 
If not, then lower case developer to match the rest.

File: src/main/resources/scripts/simple join and leave message.sk (hunk 8 above)
Comment (Matocolotoe) [reply src/main/resources/scripts/simple join and leave message.sk]: Skript handles both but at least if you want to use one in particular, use it everywhere

File: docs/src/utils/tutorials.json [hunk 11]
```diff
@@ -0,0 +1,14 @@
...
+        "description": "A fast tutorial about texts in skript."
+    },
+    "Loops": {
+        "description": "What is a loop? And how make loops in Skript? Look here!"
+    },
+    "Functions": {
+        "description": "The most usefull thing in any language: functions!"
+    },
+    "Custom Commands": {
+        "description": "The most usefull thing in Skript: custom commands!"
```
Comment (TPGamesNL) [inline docs/src/utils/tutorials.json]:
```suggestion
        "description": "The most useful thing in Skript: custom commands!"
```

File: src/main/resources/scripts/simple join and leave message.sk (hunk 8 above)
Comment (Olyno) [reply src/main/resources/scripts/simple join and leave message.sk]: Yeah I missed to change the 2nd text with ``string``. I will fix that.

File: docs/README.md (hunk 6 above)
Comment (bensku) [reply docs/README.md]: At least React supports rendering static content on server. Most frameworks do, though I don't know about Svelte.

File: docs/README.md (hunk 5 above)
Comment (bensku) [reply docs/README.md]: I'll clarify this a bit: currently, on Linux, Windows and Mac, Skript needs JDK 8 or newer to be compiled. Gradle wrapper provides Gradle, which in turn provides all compile-time dependencies. So, only JDK needed to produce a usable jar.

Spigot installation is, at the moment, needed for generating documentation. We could probably get rid of this, though, and just prepare classpath with dependencies (Guava, etc.) in build script.

File: docs/README.md (hunk 6 above)
Comment (Olyno) [reply docs/README.md]: Svelte too. Svelte is compiled using a bundler (as webpack or my favorite, rollup) into a static js file, what the website is using.

Comment (Pikachu920) [reply docs/README.md]: in this context, static doesn't mean that the files are the same every time - it means that the site has no dynamic content (like DOM manipulation and such via JS). The reason we don't want JS is because it just isn't necessary - it increases page load and parse times for no real benefit for a site as simple as Skript's docs.

Comment (Olyno) [reply docs/README.md]: I'm agree with you, but we could add more features later, that's the reason of my choice.

File: docs/README.md (hunk 5 above)
Comment (Olyno) [reply docs/README.md]: Documentation should be not related to any minecraft server

Comment (Pikachu920) [reply docs/README.md]: the doc site is generated within the plugin itself, so it runs where the plugin runs (a minecraft server)

File: docs/README.md (hunk 6 above)
Comment (bensku) [reply docs/README.md]: Web browsers are pretty good at caching static pages too. What I'd like to see is static pages with dynamic features implemented in JS. If that is not feasible, static pages when JS is not enabled, otherwise everything done with JS. If even that can't be done, two sites for browsing docs wouldn't be any worse than current situation. In fact, I suggested the last option when you were initially working on this PR.

**Comment by Olyno** [comment]
I close this pull request to split it better and make it more easily to review. Thanks to everyone who reviewed this pull request.
